src/org/mozilla/javascript/NativeJSON.java

```diff
@@ -374,12 +388,12 @@ private static String jo(Scriptable value, StringifyState state) {
         state.indent = state.indent + state.gap;
         Object[] k = null;
         if (state.propertyList != null) {
-            k = state.propertyList.toArray();
```

Comment:
`propertyList.toArray()` is called for every object and child object being stringified where propertyList is a LinkedList.